code/data_structures/src/queue/dequeue/dequeue.java

```diff
@@ -0,0 +1,116 @@
+//  data_structures|queue|dequeue|Java
+//  Part of Cosmos by OpenGenus Foundation
+
+// Implementation using doubly linked list
+
+public class Dequeue {
+    
+    private Node front = null;
+    private Node rear = null;
+    
+    public void insertAtFront(int d) {
+        Node n = new Node(d);
+        n.setNext(front);
+        
+        if (front == null) {  
+            front = n;
+            rear = n;
+        } else {
+            front.setPrevious(n);
+            front = n;
+        }
+    }
+    
+    public void insertAtRear(int d) {
+        Node n = new Node(d);
+
+        if (rear == null) {
+            front = n;
+            rear = n;
+        } else {
+            rear.setNext(n);
+            n.setPrevious(rear);
+            rear = n;
+        }
+    }
+    
+    public int removeFront() {
+        try {
+            int frontValue = front.getValue();
+            Node newFront = front.getNext();
+            if (newFront != null)  {
+                newFront.setPrevious(null);
+            }
+            front = newFront;
+            return frontValue;
+        }
+        catch (Exception e) {
```

Comment:
I disagree with this approach. Throw an exception for an empty container, as that is the idiomatic way to do things. Don't print to the console.